catalog/plugin/catalog-plugin-defaultvalues/src/main/java/ddf/catalog/plugin/defaultvalues/DefaultValuesPlugin.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ */
+package ddf.catalog.plugin.defaultvalues;
+
+import java.util.Map;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.DefaultAttributeValueRegistry;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreIngestPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+public class DefaultValuesPlugin implements PreIngestPlugin {
+    private final DefaultAttributeValueRegistry registry;
+
+    public DefaultValuesPlugin(DefaultAttributeValueRegistry registry) {
+        this.registry = registry;
+    }
+
+    private void setDefaultValues(Metacard metacard) {
+        // TODO (jrnorth) - What if the attribute is there but has a null value?
+        final String metacardTypeName = metacard.getMetacardType()
+                .getName();
+        registry.getAttributesWithDefaults(metacardTypeName)
+                .ifPresent(attributesWithDefaults -> {
+                    attributesWithDefaults.forEach(attributeName -> {
+                        final Attribute attribute = metacard.getAttribute(attributeName);
+                        if (hasNoValue(attribute)) {
+                            registry.getDefaultValue(metacardTypeName, attributeName)
+                                    .ifPresent(defaultValue -> metacard.setAttribute(new AttributeImpl(
+                                            attributeName,
+                                            defaultValue)));
+                        }
+                    });
+                });
+    }
+
+    private boolean hasNoValue(Attribute attribute) {
+        if (attribute != null) {
+            return attribute.getValue() == null;
+        }
+
+        return true;
+    }
```

Comment:
I prefer the shorter:

return attribute != null ? attribute.getValue() == null : true;